### PR TITLE
companion ad support

### PR DESCRIPTION
**Pull request by alex-phillips**
I know it was stated in an issue (https://github.com/MailOnline/videojs-vast-vpaid/issues/82) that companion ads were not supported nor were they on the roadmap, and this PR may not integrate companions to the level of other plugins, but this will at least process and expose companion ads to the developer on a basic level to allow them to handle them as they see fit.

It adds the `Companion` class (with tests) as well as attaching the `VASTResponse` object to `player.vast` so that the dev can listen on the event `vast.adStart` and have access to the response which, if they choose, they can traverse and find any companion ads they may want to utilize.

Companion ads are a BIG part of VAST ads and this plugin should have a way to at least access that information.

### CODE REVIEW

File: test/ads/vast/client/Companion.spec.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      assert.equal(companion.creativeResource, "http://www.example.com/image.jpg");
+      assert.equal(companion.creativeType, "image/jpeg");
+    });
+  });
+
+  //Optional Elements
+  describe("trackingEvents", function(){
+    /*jshint maxlen: 500 */
+    it("must be filled with all the tracking events that the xml provides", function(){
+      var companionXML = '<Companion>' +
```
Comment (massa84) [inline test/ads/vast/client/Companion.spec.js]: cant you put this xml (that can be shared across all test) in an external variable and simplify the creativeView url ?
something like 
 
```javascript
var companionXML = '<Companion width="300" height="250">' +
    '<StaticResource creativeType="image/jpeg"><![CDATA[http://www.example.com/image.jpg]]></StaticResource>' +
    '<TrackingEvents><Tracking event="creativeView"><![CDATA[http://www.example.com/whatever]]></Tracking></TrackingEvents>' +
    '<CompanionClickThrough><![CDATA[http://www.example.com/tracker.html]]></CompanionClickThrough>' 
'</Companion>';
```

Comment (alex-phillips) [inline test/ads/vast/client/Companion.spec.js]: @massa84 Definitely, I just used the inline XML example from `Linear.js` as a guide to build the test and used the same tracking URL. Where would you want the external XML file to be?

Comment (alex-phillips) [inline test/ads/vast/client/Companion.spec.js]: @massa84 This is done. Let me know of any other changes.

**Comment by alex-phillips** [comment]
Any update on this?

**Comment by alex-phillips** [comment]
I'm going to close this one and replace it with the `videojs-5x` branch version.

**Pull request closed without merging**